src/renderer/page/subscriptions/view.jsx

```diff
@@ -43,69 +32,45 @@ export default class extends React.PureComponent<Props> {
     setSubscriptionNotifications(newNotifications);
   }
 
-  componentWillReceiveProps(props: Props) {
-    const { savedSubscriptions, hasFetchedSubscriptions, setHasFetchedSubscriptions } = props;
+  componentDidUpdate() {
+    const {
+      subscriptions,
+      subscriptionClaims,
+      doFetchClaimsByChannel,
+      subscriptionsBeingFetched,
+    } = this.props;
 
-    if (!hasFetchedSubscriptions && savedSubscriptions.length) {
-      this.fetchSubscriptions(savedSubscriptions);
-      setHasFetchedSubscriptions();
-    }
-  }
+    const subscriptionClaimMap = {};
+    subscriptionClaims.forEach(claim => {
+      subscriptionClaimMap[claim.uri] = 1;
+    });
 
-  fetchSubscriptions(savedSubscriptions: SavedSubscriptions) {
-    const { doFetchClaimsByChannel } = this.props;
-    if (savedSubscriptions.length) {
-      // can this use batchActions?
-      savedSubscriptions.forEach(sub => {
+    subscriptions.forEach(sub => {
+      if (!subscriptionClaimMap[sub.uri] && !subscriptionsBeingFetched[sub.uri]) {
         doFetchClaimsByChannel(sub.uri, 1);
-      });
-    }
+      }
+    });
   }
 
   render() {
-    const { subscriptions, savedSubscriptions } = this.props;
-
-    // TODO: if you are subscribed to an empty channel, this will always be true (but it should not be)
-    const someClaimsNotLoaded = Boolean(
-      subscriptions.find(subscription => !subscription.claims.length)
-    );
+    const { subscriptions, subscriptionClaims, isFetchingSubscriptions } = this.props;
 
-    const fetchingSubscriptions =
-      !!savedSubscriptions.length &&
-      (subscriptions.length !== savedSubscriptions.length || someClaimsNotLoaded);
+    let claimList = [];
+    subscriptionClaims.forEach(claimData => {
```

Comment:
Good point, it should just be in a selector

Reply:
Actually upon closer look I think it's ok if it's inside `render()`. I kept it inside this in this format so we can check if a subscription uri has any fetched claims in the `componentDidUpdate`

Since we are already passing them in this format, I don't think it makes sense to create a separate selctor to pass in the same data in a different format (I think)

Reply:
:+1: